File: vscodeExt/283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = 0 #[0,k) are non-zeroes
        i = 0 
        while i < len(nums):
            if nums[i] :
                nums[k] = nums[i]
                k += 1
            
            i+=1    
        while k < len(nums):
            nums[k]  = 0 
            k+=1

# @lc code=end

File: vscodeExt/283.move-zeroes.java
/*
 * @lc app=leetcode id=283 lang=java
 *
 * [283] Move Zeroes
 */

// @lc code=start
class Solution {
    public void moveZeroes(int[] nums) {
        // [0,k) are non-zeros
        int k = 0;
        // i current index
        int i = 0; 
        for(;i<nums.length;i++){
            if( nums[i] !=0){
                nums[k] = nums[i];
                k+=1;
            }
        }
        for(;k<nums.length;k++){
            nums[k] = 0;
        }
    }
}
// @lc code=end

